framework/src/dirackernels/ReporterPointSource.C

```diff
@@ -0,0 +1,88 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ReporterPointSource.h"
+#include "MooseUtils.h"
+#include "DelimitedFileReader.h"
+
+registerMooseObject("MooseApp", ReporterPointSource);
+
+defineLegacyParams(ReporterPointSource);
+
+InputParameters
+ReporterPointSource::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+
+  params.addClassDescription("Apply a point load defined by Reporter.");
+
+  params.addRequiredParam<std::string>(
+      "x_coord_name",
+      "reporter x-coordinate name.  This uses the reporter syntax <reporter>/<name>.");
+  params.addRequiredParam<std::string>(
+      "y_coord_name",
+      "reporter y-coordinate name.  This uses the reporter syntax <reporter>/<name>.");
+  params.addRequiredParam<std::string>(
+      "z_coord_name",
+      "reporter z-coordinate name.  This uses the reporter syntax <reporter>/<name>.");
+  params.addRequiredParam<std::string>(
+      "value_name", "reporter value name.  This uses the reporter syntax <reporter>/<name>.");
+
+  return params;
+}
+
+ReporterPointSource::ReporterPointSource(const InputParameters & parameters)
+  : DiracKernel(parameters),
+    ReporterInterface(this),
+    _values(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("value_name")), REPORTER_MODE_REPLICATED)),
+    _x_coord(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("x_coord_name")), REPORTER_MODE_REPLICATED)),
+    _y_coord(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("y_coord_name")), REPORTER_MODE_REPLICATED)),
+    _z_coord(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("z_coord_name")), REPORTER_MODE_REPLICATED))
```

Comment:
With `addRequiredParam<ReporterName>` you just have to do `_values(getReporterValue("value_name")),`

Reply:
How can check that they are in REPORTER_MODE_REPLICATED

Reply:
Sorry, should be `_values(getReporterValue("value_name", REPORTER_MODE_REPLICATED)),